### PR TITLE
Clear separation of API schemas (request and respponse DTOs) and data models, lower case comparison of image type name for case insensitive fetch/search, refactoring and fixes (Jira: BDP-4381)).

### PR DESCRIPTION
**Pull request by abhisheknath2011**
Jira: BDP-4381
Included changes
1. Clear separation of API schemas (request and response DTOs) and data models. Even though, there are one to one correspondence between two in most of the cases, the change would enable more extensibility in future such as independently adding/removing different fields API DTOs and backend models etc.
2. Lowercase comparison of image type name for case insensitive fetch/search.
3. Code refactoring and cleanup.
4. Renaming changes.
5. Fixed bugs found during testing.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDao.java [hunk 1]
```diff
@@ -36,7 +35,7 @@
    * @param imageRampupPlanRequest
    * @return int - id of the image version
    */
-  public int createImageRampupPlan(ImageRampupPlanRequest imageRampupPlanRequest)
+  public int createImageRampupPlan(ImageRampupPlan imageRampupPlanRequest)
```
Comment (arvindpruthi) [inline azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDao.java:38]: Nit: I think the keyword "Request" should be removed from the variable names.

File: azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDaoImpl.java [hunk 2]
```diff
@@ -110,26 +115,23 @@ public ImageRampupDaoImpl(final DatabaseOperator databaseOperator,
...
+   * associated image version, rampup percentage etc. in the image_rampup table. Also check if the
+   * specified version is valid and exists. Otherwise, fail the whole transaction. Here is the
+   * example input for ImageRampupPlanRequest { "planName": "Rampup plan 1", "imageType":
+   * "spark_job", "description": "Ramp up for spark job", "activatePlan": true, "imageRampups": [ {
+   * "imageVersion": "1.6.2", "rampupPercentage": "70" }, { "imageVersion": "1.6.1",
+   * "rampupPercentage": "30" } ] }
    */
   @Override
-  public int createImageRampupPlan(final ImageRampupPlanRequest imageRampupPlanRequest) {
+  public int createImageRampupPlan(final ImageRampupPlan imageRampupPlanRequest) {
```
Comment (arvindpruthi) [inline azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDaoImpl.java:134]: Nit: Remove "Request" from the variable name ?

File: azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDaoImpl.java [hunk 3]
```diff
@@ -315,14 +325,14 @@ public int createImageRampupPlan(final ImageRampupPlanRequest imageRampupPlanReq
   }
 
   @Override
-  public void updateImageRampupPlan(final ImageRampupPlanRequest imageRampupPlanRequest)
+  public void updateImageRampupPlan(final ImageRampupPlan imageRampupPlanRequest)
```
Comment (arvindpruthi) [inline azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDaoImpl.java:328]: Nit: Remove "Request" keyword from the variable name.

File: azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java [hunk 4]
```diff
@@ -207,23 +224,34 @@ private static void setupImageTables() {
     imageTypeDao = new ImageTypeDaoImpl(dbOperator);
     imageVersionDao = new ImageVersionDaoImpl(dbOperator, imageTypeDao);
     imageRampupDao = new ImageRampupDaoImpl(dbOperator, imageTypeDao, imageVersionDao);
-    imageRampupManager = new ImageRampupManagerImpl(imageRampupDao, imageVersionDao,
-        imageTypeDao);
+    // Create a mock of ImageRampupManager to get the image type and version map. This mock is
+    // required as the completed flow of getting image type and version can't be tested by
+    // populating image management table due non supported "UNSIGNED" integer in HSQL.
+    /*imageRampupManager = new ImageRampupManagerImpl(imageRampupDao, imageVersionDao,
```
Comment (arvindpruthi) [inline azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java:230]: Remove the commented/dead code.

**Review by arvindpruthi — APPROVED** [review]

**Comment by abhisheknath2011** [comment]
Thanks @arvindpruthi for reviewing. Will take care of the nit comment separately and merging this PR.

**Pull request merged**